Tools/CMake/AMReX_Options.cmake

```
@@ -206,16 +206,19 @@ endif ()
 option( ENABLE_PIC "Build position-independent code" OFF)
 print_option( ENABLE_PIC )
 
-option(ENABLE_FPE "Enable Floating Point Exceptions checks" OFF)
-print_option( ENABLE_FPE )
-
+set(ENABLE_FPE_DEFAULT OFF)
+set(ENABLE_ASSERTIONS_DEFAULT OFF)
 if ( "${CMAKE_BUILD_TYPE}" MATCHES "Debug" )
-   option( ENABLE_ASSERTIONS "Enable assertions" ON)
-else ()
-   option( ENABLE_ASSERTIONS "Enable assertions" OFF)
+   set(ENABLE_ASSERTIONS_DEFAULT ON)
+   set(ENABLE_FPE_DEFAULT ON)
 endif ()
+cmake_dependent_option(ENABLE_ASSERTIONS "Enable assertions" ENABLE_ASSERTIONS_DEFAULT
+    "ON" ENABLE_ASSERTIONS_DEFAULT)
+cmake_dependent_option(ENABLE_FPE "Enable Floating Point Exceptions checks" ENABLE_FPE_DEFAULT
+    "ON" ENABLE_FPE_DEFAULT)
 
 print_option( ENABLE_ASSERTIONS )
+print_option( ENABLE_FPE )
```

```suggestion
cmake_dependent_option( ENABLE_ASSERTIONS "Enable assertions" ON
   "CMAKE_BUILD_TYPE MATCHES Debug" OFF)
print_option( ENABLE_ASSERTIONS )
```
I think this is cleaner and uses cmake_dependent_options as it was intended to be used.

---

I think this hides the option for all non-debug builds.